tests/rptest/tests/consumer_group_test.py

```diff
@@ -1505,3 +1506,192 @@ def verify_all_consumers_are_present():
+        ]
+        cg_tp = ConsumerGroupTopicPartitions(self.group_id,
+                                             topic_partitions=topic_partitions)
+        offsets = self.admin_client.list_consumer_group_offsets([cg_tp])
+
+        return offsets[self.group_id].result()
+
+    def total_committed(self):
+        return sum([c.get_last_committed() for c in self.consumers])
+
+    def wait_for_total_commits(self, total_commits):
+        last_total = 0
+        while self.total_committed() < total_commits:
+            wait_until(
+                lambda: self.total_committed() > last_total, 60, 2,
+                "Timeout waiting for consumers to make progress committing offsets"
+            )
+            last_total = self.total_committed()
+            self.logger.debug(f"Total offsets committed: {last_total}")
+            time.sleep(5)
+
+    @cluster(num_nodes=3)
+    @skip_debug_mode
+    def test_stress_consumer_group_commits(self):
+        self.consumer_count = 200
+
+        topic = TopicSpec(name="cg-test-topic-1",
+                          partition_count=self.consumer_count)
+        DefaultClient(self.redpanda).create_topic(topic)
+        self.group_id = "test-group-1"
+
+        self.consumers: list[OffsetCommitter] = []
+
+        for c_id in range(self.consumer_count):
+            self.consumers.append(
+                OffsetCommitter(
+                    bootstrap_servers=self.redpanda.brokers(),
+                    group=self.group_id,
+                    topic=topic.name,
+                    id=c_id,
+                    logger=self.logger,
+                    partition_id=c_id,
+                ))
+
+        self.admin_client = AdminClient(
+            {"bootstrap.servers": self.redpanda.brokers()})
+        total_commits = 2_000_000
+        self.wait_for_total_commits(total_commits)
+
+        rp_admin = Admin(self.redpanda)
+        for i in range(3):
+            # transfer leadership of __consumer_offsets to a different node
+            rp_admin.partition_transfer_leadership("kafka",
+                                                   "__consumer_offsets", 0)
+            time.sleep(2)
+
+        wait_until(
+            lambda: self.get_group_description().state == ConsumerGroupState.
+            EMPTY,
+            timeout_sec=60,
+            backoff_sec=1,
+            retry_on_exc=True,
+        )
+
+        for c in self.consumers:
+            c.stop()
+
+        # list committed offsets
+        gd = self.list_consumer_group_offsets(topic.name)
+
+        for tp in gd.topic_partitions:
+            expected = self.consumers[tp.partition].get_last_committed()
+            self.logger.info(
+                f"Partition {tp.topic}/{tp.partition} committed offset={tp.offset}"
+            )
+            assert tp.offset == expected, f"Offset mismatch for partition {tp.topic}{tp.partition}, expected: {expected} got: {tp.offset}"
+
+        olv = OfflineLogViewer(self.redpanda)
```

Comment:
also think we could reboot the cluster and do the list_offsets check again (drops all cached state).

Reply:
i was thinking about parametrising this test with the disturbance type